core/src/main/java/org/apache/iceberg/TableProperties.java

```diff
@@ -231,15 +231,8 @@ private TableProperties() {}
   public static final String OBJECT_STORE_ENABLED = "write.object-storage.enabled";
   public static final boolean OBJECT_STORE_ENABLED_DEFAULT = false;
 
-  /** @deprecated Use {@link #WRITE_DATA_LOCATION} instead. */
```

Comment:
Ah, I think this was misleading. This is telling people not to use the property, but it doesn't mean that we can remove the property. As I said in a comment above, this will break older tables so there's no good reason to ever remove it.